inji-web/src/pages/User/Passcode/PasscodePage.tsx

```
@@ -69,58 +93,51 @@ export const PasscodePage: React.FC = () => {
+            handleWalletStatusError(errorCode, true);
+            throw response.error;
         }
+        saveWalletId(walletId)
     };
 
     const createWallet = async () => {
-        try {
-            const pin = passcode.join('');
-            const confirmPin = confirmPasscode.join('');
-
-            if (pin !== confirmPin) {
-                setError(t('error.passcodeMismatchError'));
-                throw new Error('Pin and Confirm Pin mismatch');
-            }
+        const pin = passcode.join('');
+        const confirmPin = confirmPasscode.join('');
 
-            const response = await createWalletApi.fetchData({
-                apiConfig: api.createWalletWithPin,
-                body: {
-                    walletPin: pin,
-                    confirmWalletPin: confirmPasscode.join(''),
-                    walletName: null
-                },
-            })
+        if (pin !== confirmPin) {
+            setError(t('error.passcodeMismatchError'));
+            throw new Error('Pin and Confirm Pin mismatch');
+        }
 
-            if (!response.ok()) {
-                const errorMessage = ((response.error as ApiError)?.response?.data as ErrorType).errorMessage ?? t('unknown-error');
-                setError(
-                    `${t('error.createWalletError')}: ${errorMessage}`
-                );
-                throw response.error;
-            }
+        const response = await createWalletApi.fetchData({
+            apiConfig: api.createWalletWithPin,
+            body: {
+                walletPin: pin,
+                confirmWalletPin: confirmPasscode.join(''),
+                walletName: null
+            },
+        })
+
+        if (!response.ok()) {
+            const errorMessage = ((response.error as ApiError)?.response?.data as ErrorType).errorMessage ?? t('unknown-error');
+            setError(
+                `${t('error.createWalletError')}: ${errorMessage}`
+            );
+            throw response.error;
```

same here Do we need to throw this error explicitly to caller given error handling (zetError) is already done and the caller is not necessarily listening to the error thrown?